tests/test_sftp_client.cpp

```diff
@@ -77,10 +77,19 @@ struct SFTPClient : public testing::Test
         return {std::make_unique<mp::SSHSession>("b", 43, "ubuntu", key_provider)};
     }
 
+// this is a macro since REPLACE only applies to the current scope and cannot be moved out.
```

Comment:
Oh you need the capture so extracting the lambda wouldn't work. You'd need `std::bind` or another lambda again. Alright.